File: bankingonrails/app/models/externalaccount.rb
class ExternalAccount < ApplicationRecord


  composed_of :i_b_a_n,
    class_name: "IBAN",
    mapping: [
      %w[i_b_a_n_value value]
    ]

  composed_of :account_number,
    class_name: "AccountNumber",
    mapping: [
      %w[account_number_value value]
    ]

  composed_of :b_i_c,
    class_name: "BIC",
    mapping: [
      %w[b_i_c_value value]
    ]

  has_many :Customer, class_name: 'Customer'
  has_many :Transactions, class_name: 'Transaction'

end

File: bankingonaspdotnet/Domain/ExternalAccount.cs

using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class ExternalAccount
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public virtual long? ExternalaccountId { get; set; }
    public virtual string? Name { get; set; }
    public virtual IBAN? Iban { get; set; }
    public virtual AccountNumber? AccountNumber { get; set; }
    public virtual BIC? Bic { get; set; }
    public virtual string? BankName { get; set; }
    public virtual string? Country { get; set; }
    public virtual Customer? Customer { get; set; }
    public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();

    public static ExternalAccount FromRequest(ExternalAccountRequest request)
    {
        return new ExternalAccount
        {
            Id = request.Id,
            Name = request.Name,
            Iban = request.Iban,
            AccountNumber = request.AccountNumber,
            Bic = request.Bic,
            BankName = request.BankName,
            Country = request.Country,
        };
    }
}
